### PR TITLE
Enable variable modifications to layer changes on the same variable

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge allows modifiers that modify the same variable to layer their changes onto the modified variable.

Previously, only the last modifier to modify a variable would have it's modification stick.

### CODE REVIEW

File: lib/ramble/ramble/modifier.py [hunk 1]
```diff
@@ -255,10 +255,13 @@ def modded_variables(self, app):
                 # var_str = app.expander.expansion_str(var)
                 # prev_val = app.expander.expand_var(var_str)
                 prev_val = app.variables[var]
+                if var in extra_vars:
```
Comment (linsword13) [inline lib/ramble/ramble/modifier.py:255]: nit: maybe using the single-assignment-form is clearer?

```
if var in extra_vars:
    prev_val = extra_vars[var]
else:
    prev_val = app.variables[var]
```

**Review by linsword13 — COMMENTED** [review]
LGTM. Do you think it'd be good to add a test for it?

File: lib/ramble/ramble/modifier.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/modifier.py:255]: Yeah, good point. Fixed.

**Comment by douglasjacobsen** [comment]
I added a test, and updated existing modifiers to include the space separator as part of their variable modifications (given I removed the auto-injected space)

**Comment by linsword13** [comment]
> I added a test, and updated existing modifiers to include the space separator as part of their variable modifications (given I removed the auto-injected space)

Do you feel having the separator auto-injected would be less error-prone than relying on modifier author to remember adding the space?

Perhaps a more general approach is to have the directive to accept a `template_func` (`(prev_value, new_mod) => new_value`)? (More just for a discussion, no need to add to the current PR though.)

**Comment by douglasjacobsen** [comment]
Yes, I actually have an internal task tracking this which I think is a high priority.

Injecting a space manually is definitely a hack for the time being, and we should have a nicer interface for this.

**Review by linsword13 — APPROVED** [review]

**Pull request merged**